tsl/src/nodes/gapfill/gapfill_exec.c

```diff
@@ -1287,6 +1271,40 @@ gapfill_state_initialize_columns(GapFillState *state)
 			continue;
 		}
 
+		/*
+		 * For every column with Aggrefs we take the original expression tree from the
+		 * subplan and replace Aggref nodes with Const NULL nodes. This is
+		 * necessary because the expression might be evaluated below the
+		 * aggregation so we need to pull up expression from subplan into
+		 * projection for gapfilled tuples so expressions like COALESCE work
+		 * correctly for gapfilled tuples.
+		 */
+		if (column_contains_aggs)
+		{
+			Node *entry = copyObject((Node *) tle);
+			entry = gapfill_aggref_mutator(entry, NULL);
+
+			/* Fix for #4894 when we have expressions like (agg + gby_var):
+			 * after replacing aggs with NULLs recheck column for gby vars,
+			 * fix those gby vars to refer to exec vars and mark the column DERIVED
+			 */
+			if (contain_var_clause(entry))
+			{
+				gapfill_fix_gby_vars(entry, cscan->custom_scan_tlist, exec_tlist);
```

Comment:
I think the "gby" abbreviation is not used anywhere else, do you mind expanding it to something like "grouping" so that it's easier to find? E.g. this file uses "GROUP_COLUMN" and so on.